integration-tests/deployment/ccip/add_lane_test.go

```diff
@@ -2,31 +2,62 @@ package ccipdeployment
 
 import (
 	"testing"
+	"time"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/require"
 
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
 	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+
 	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
 )
 
 // TestAddLane covers the workflow of adding a lane
 // between existing supported chains in CCIP.
-func TestAddLane(t *testing.T) {
-	// TODO: The offchain code doesn't yet support partial lane
-	// enablement, need to address then re-enable this test.
-	t.Skip()
-	e := NewMemoryEnvironmentWithJobs(t, logger.TestLogger(t), 3, 4)
+func TestAddAndDisableLane(t *testing.T) {
+	// We add more chains to the chainlink nodes than the number of chains where CCIP is deployed.
+	e := NewMemoryEnvironmentWithJobs(t, logger.TestLogger(t), 4, 4)
 	// Here we have CR + nodes set up, but no CCIP contracts deployed.
 	state, err := LoadOnchainState(e.Env, e.Ab)
 	require.NoError(t, err)
+
+	selectors := e.Env.AllChainSelectors()
+	// deploy CCIP contracts on two chains
+	chain1, chain2 := selectors[0], selectors[1]
+
+	feeds := state.Chains[e.FeedChainSel].USDFeeds
+	tokenConfig := NewTokenConfig()
```

Comment:
drive by - can we make this constructor just accept a map of token info vs upserting everywhere?

Reply:
Honestly why is this even needed? Would prefer just using the offchain config types directly instead of this wrapper